File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    // 增加品牌
    this.add = function (entity) {
        return $http.post("../brand/save.do", entity);
    };

    // 修改品牌
    this.findById = function (id) {
        return $http.get("../brand/findById.do?id=" + id);
    };
    this.update = function (entity) {
        return $http.post("../brand/updateBrand.do", entity);
    };

    // 删除品牌
    this.dele = function (ids) {
        return $http.post("../brand/deleteBrand.do", ids);
    };

    // 模糊查询加分页请求
    this.search = function (currentPage, pageSize, searchEntity) {
        return $http.post("../brand/findPage.do?currentPage=" + currentPage + "&pageSize=" + pageSize, searchEntity);
    };

    // 返回品牌信息json字符串[{id:1,text:“华为”},{},...]
    this.selectOptionList = function () {
        return $http.get("../brand/selectOptionList.do");
    }

});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.github.pagehelper.PageInfo;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {
    /**
     * 查询所有品牌
     *
     * @return List<TbBrand>
     */
    List<TbBrand> findAll();

    /**
     * 品牌分页
     *
     * @param currentPage 当前页码
     * @param pageSize    每页记录数
     * @return PageInfo对象
     */
    PageInfo findByPage(Integer currentPage, Integer pageSize);

    /**
     * 添加品牌
     *
     * @param TbBrand
     */
    void save(TbBrand tbBrand);

    /**
     * 根据id查询品牌信息
     *
     * @param Integer
     * @return TbBrand
     */
    TbBrand findById(Long id);

    /**
     * 修改品牌信息
     *
     * @param TbBrand
     */
    void updateBrand(TbBrand tbBrand);

    /**
     * 根据id删除品牌
     *
     * @param Long[]
     */
    void deleteBrand(Long[] ids);

    /**
     * 根据条件查询品牌信息
     * @param TbBrand
     * @param Integer
     * @param Integer
     * @return PageInfo
     */
    PageInfo findByPage(TbBrand tbBrand, Integer currentPage, Integer pageSize);

    /**
     * 返回所有品牌信息
     * @return [{id:1,text:“华为”},{},...]
     */
    List<Map> selectOptionList();

}
